deep-learning/_posts/2018-07-27-SFCNN.md

```diff
@@ -0,0 +1,86 @@
+---
+
+
+# Contribution
+
+Observing that regular CNNs exhibit translational invariance but suffer from rotational variance, the authors
+propose a CNN architecture based on steerable filters (called *SFCNN*) featuring rotation equivariance. Employing their approach
+circumvents the need for rotation in data augmentation and reduces the '*sample complexity*' of the network.
+The authors' main argument in support of equivariance properties of a network is that by this, the hypothesis
+space (i.e. the possible transformations found in the data) is reduced which simplifies the learning task. 
+
+<p style="text-align:center"><img src="/deep-learning/images/SFCNN/overview.png" width="700"></p>
+
+
+# Steerable Filters
+
+ * A filter $$\Psi$$ is rotationally steerable if "*its rotation by an arbitrary angle $$\theta$$ can
+   be expressed*" in terms of rotations applied to a set of atomic basis functions
+   $$\{\psi_q\}^Q_{q=1}$$ resulting in
+   
+     $$\rho_\theta\Psi(x) = \sum^Q_{q=1} \kappa_q(\theta)\psi_q(x)$$.
+     
+ * The authors choose circular harmonics as the atomic basis functions:
+     
+     $$\psi_q \equiv \psi_{jk} = \tau_j(r) e^{\mathrm{i}k\phi}$$.
+     
+ * A specific learned filter is obtained by linear combination of these basis functions using weights
+   $$w_{jk} \in \mathcal{C}$$ ($$\mathcal{C}$$ denotes the complex numbers ;) ) learned by the network:
+     
+     $$\tilde{\Psi}(x) = \sum^J_{j=1} \sum^{K_j}_{k=0} w_{jk}\psi_{jk}(x)$$.
+ 
+   By choosing the learned weights to be complex, they induce rotations of the basis functions
+   with respect to each other.
+
+
+# Network architecture
+
+ * Output features on intermediate layers are obtained by so-called **group convolutions** which are a natural
+   extension of spatial convolutions to "*general transformation groups*". The group convolution is obtained
+   by "*a spatial convolution, rotation and linear combination*" (denoted $$\mathcal{R}_\phi$$).
+
+<p style="text-align:center"><img src="/deep-learning/images/SFCNN/architecture.png" width="700"></p>
+
+
+# Experiments
+
+#### Rotated MNIST
+
+<p style="text-align:center"><img src="/deep-learning/images/SFCNN/angular_discretisation.png" width="700"></p>
+
+ * **Left** - The test error saturates with for more than 12 filter rotations. Also increasing the number of training
+   samples improves performance as expected. The authors point out that an increase from 2 to 4 discrete filter rotations
+   gives the same improvement as doubling the number of training samples. This supports their argument of
+   a limited hypothesis space.
+ * **Right** - Training on conventional MNIST, testing on samples rotated by the particular angle on the horizontal
+   axis. SFCNN with 16 rotation angles.
+   
+    * **a**. Even with augmentation the CNNs do not achieve as good performance as SFCNNs.
+    * **b**. Augmentation leads to decrease of performance also for testing on unrotated data sets for CNNs,
```

Comment:
"decrease of error rate" OR "improvement in performance". Why also?

Reply:
It is actually a decrease of performance, 'even' in the case of samples that are not rotated when using conventional CNNs. This is another argument in support of SFCNNs here.